blockchain/scriptval.go

```diff
@@ -181,10 +182,25 @@ func newTxValidator(utxoView *UtxoViewpoint, flags txscript.ScriptFlags, sigCach
 
 // ValidateTransactionScripts validates the scripts for the passed transaction
 // using multiple goroutines.
-func ValidateTransactionScripts(tx *dcrutil.Tx, utxoView *UtxoViewpoint, flags txscript.ScriptFlags, sigCache *txscript.SigCache) error {
+func ValidateTransactionScripts(tx *dcrutil.Tx, utxoView *UtxoViewpoint,
+	flags txscript.ScriptFlags, sigCache *txscript.SigCache,
+	isAutoRevocationsEnabled bool) error {
+
+	// Skip version 2 revocations if the automatic ticket revocations agenda is
+	// active.  This is allowed since consensus rules enforce that revocations
+	// MUST pay to the address specified by the original commitment in the
+	// ticket.
+	msgTx := tx.MsgTx()
+	if isAutoRevocationsEnabled &&
+		stake.IsSSRtx(msgTx, isAutoRevocationsEnabled) &&
+		msgTx.Version == stake.TxVersionAutoRevocations {
```

Comment:
It wouldn't cause a problem since the updated consensus rules also enforce that the revocation transaction version must be `2`.  However, I think it would be a bit misleading to anyone reading the code to do `>=` since transactions with a version greater than `stake.TxVersionAutoRevocations` aren't even allowed.